content/en/blog/2022/opamp/index.md

```diff
@@ -0,0 +1,239 @@
+
+import (
+   "fmt"
+   "gopkg.in/yaml.v3"
+   "io/ioutil"
+   "log"
+   "net/http"
+   "path/filepath"
+)
+
+type configurations struct {
+   Instrument bool
+}
+
+func httpHandler(w http.ResponseWriter, r *http.Request) {
+   fmt.Fprintf(w, "Hi! This action could create a trace!")
+}
+
+func main() {
+   filename, _ := filepath.Abs("./effective.yaml")
+   yamlFile, _ := ioutil.ReadFile(filename)
+
+   var config configurations
+   yaml.Unmarshal(yamlFile, &config)
+
+   handler := http.HandlerFunc(httpHandler)
+   http.Handle("/", handler)
+   fmt.Println("Starting server on port 8080")
+   log.Fatal(http.ListenAndServe(":8080", nil))
+}
+```
+
+Sorry, no error handling here. Next, let's wrap our handler with instrumentation
+and condition it with our configuration file. Something like this:
+
+```go
+package main
+
+import (
+   "context"
+   "fmt"
+   "go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+   "go.opentelemetry.io/otel"
+   "go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
+   sdktrace "go.opentelemetry.io/otel/sdk/trace"
+   "go.opentelemetry.io/otel/trace"
+   "gopkg.in/yaml.v3"
+   "io/ioutil"
+   "log"
+   "net/http"
+   "os"
+   "path/filepath"
+)
+
+type configurations struct {
+   Instrument bool
+}
+
+var tracer trace.Tracer
+
+func newConsoleExporter() (sdktrace.SpanExporter, error) {
+   return stdouttrace.New(
+      stdouttrace.WithWriter(os.Stdout),
+      stdouttrace.WithPrettyPrint(),
+   )
+}
+
+func httpHandler(w http.ResponseWriter, r *http.Request) {
+   fmt.Fprintf(w, "Hi! This action could create a trace!")
+}
+
+func setHandler(handler http.Handler, config configurations) http.Handler {
+   if config.Instrument {
+      return otelhttp.NewHandler(handler, "instrumentation activated by OpAMP")
+   }
+   return http.HandlerFunc(httpHandler)
+}
+
+func main() {
+   filename, _ := filepath.Abs("./effective.yaml")
+   yamlFile, _ := ioutil.ReadFile(filename)
+
+   var config configurations
+   yaml.Unmarshal(yamlFile, &config)
+
+   exp, _ := newConsoleExporter()
+   tp := sdktrace.NewTracerProvider(sdktrace.WithBatcher(exp))
+   defer func() { _ = tp.Shutdown(context.Background()) }()
+
+   otel.SetTracerProvider(tp)
+
+   tracer = tp.Tracer("ControlledOpAMPAgentDemo")
+   handler := http.HandlerFunc(httpHandler)
+   http.Handle("/", setHandler(handler, config))
+   fmt.Println("Starting server on port 8080")
+   log.Fatal(http.ListenAndServe(":8080", nil))
+}
+```
+
+Build it with `go build .` Now if we run it and open http://localhost:8080/ on a
+browser, nothing special will be shown. It’s time to add some opamp. Git clone
+[opamp-go](https://github.com/open-telemetry/opamp-go) and run the server with:
+
+```
+cd internal/examples/server
+go run .
+```
+
+Go to http://localhost:4321/ to verify it running, notice no agent displayed:
+
+![No agents display on opamp server demo UI](opamp_server_no_agents.png)
+
+Next, open a new terminal and on internal/examples/supervisor/bin edit
```

Comment:
Please reformat this to show the commands in the terminal, rather than expressing as text. ie

Next, open a new terminal and run the following commands
```console
foo@bar: cd internal/examples/supervisor/bin 
foo@bar: edit supervisor.yaml 
```
etc

Reply:
This assumes that people edit files on the console, of course they should be able to use whatever editor they like, so go with "Edit `supervisor.yaml` in `internal/examples/supervisor/bin`. 

Give an inline yml that shows how it should look like.